project/conventions.md

```diff
@@ -0,0 +1,108 @@
+under all future circumstances.  For this reason, many conventions also
+have a small explanation of the rationale behind the guidance, allowing
+future development efforts to weigh and balance the conflicting forces
+in ambiguous situations not clearly covered.
+
+This document is intended to be a living document.  If you have an issue
+with a convention, we hope that you will create an issue describing the
+problem, and a pull request suggesting the update.  By addressing the
+convention, we hope to avoid arguments about conventions in the pull
+request review process, which slow the process and are often subject to
+the popular whims of the moment.
+
+## Conventions used in this document
+
+While this is not a software specification, the key words "MUST",
+"MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+"RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be
+interpreted as closely to the convention as it would be to a
+specification described in RFC 2119.
+
+Use of the word in its non-all caps form SHOULD NOT be interpreted in
+the strict definitions user RFC 2119, but SHOULD be interpreted in
+context with a meaning aligned to RFC 2119.
+
+## Helm Chart conventions
+
+Helm values are organized in a tree structure, with each entry having
+a key, the path within the structure, and a value, the contents held
+at that path.  Some of the contents include objects and arrays, which
+are structures that permit multiple values under a particular path.
+
+This design choice mimics the representation of a file system, and we
+may use file system analogies to describe the conventions and choices
+in the project.
+
+### Preference for absolute paths in templates
+
+Outside of looping constructs that require relative paths, templates
+should use absolute paths (paths starting with the `.` character) for
+their value references.  
+
+### Helm Key path elements
+
+Elements refer to items in the configuration.  Some of those items are
+external to the project, while others are components the project choose
+to implement the Helm Chart packaging solution.
+
+### Preferred characters for helm path elements
+
+Elements within a helm chart path SHOULD limit the use of special
+characters, even when such characters are supported by Helm.  ASCII
+letters, upper and lower case, ASCII numerals, and the ASCII underscore
+character are always permitted.  
+
+Other characters, such as `@`, `~`, `-`, `&`, `+` are often used as
+shorthand to represent, respectively `at`, `similar / not`, 
+`minus / without`, `and / reference`, `plus / with`.  Helm key elements
+MUST NOT contain such symbols, as it burdens the reader to deduce the
+intent of the writer.
+
+### Camel Case for created elements.
+Elements that refer to components that the project has created should
+exclusively use camel case, with a the first word being lower case.  To
+illustrate, a "card sort order" path element would read "cardsharper".
+
```

Comment:
We're configuring spire options. maybe some discussion about when it is appropriate to use the option as specified to spire and when to follow a stricter camel casing scheme. For example, plugin names.

Reply:
@kfox1111 there's an entire second paragraph that talks about items we don't control the naming of, like plugin names.  I think your comment is sufficiently handled there, where the answer is "make it nearly identical to an external name, including matching capitalization, only replacing spaces and hyphens with underscores (for obvious reasons of following Helm's own documented best practices)"

Reply:
This? https://github.com/spiffe/helm-charts/pull/331/files#diff-f0ed57313266c949a152fa5790e72e4b36629acf9ae273fe84ac48f779bef52eR9-R14

Its pretty vague with regard to plugin names.

I ask because of this:
https://github.com/spiffe/helm-charts/pull/227/files#diff-44e90d1febb559dfb70b0529bd6b5bc4bc0c51ed995407d2314f4aaf1062cac4L3-R3

The plugin name has a '-' in it, and would be manual code to try and map it in that case. not sure its worth one offing random plugin names into a _ form.

Reply:
I get that you are uncomfortable with the suggestion, but I don't have a clue as to what you are suggesting in it's place.  Is there a better idea you want to put on the table?  I'm all ears.

We could do "absolutely the same word, no replacements", we could permit hyphens, disregarding the Helm Chart's own published best practices, we could make - the replacement character, or we could do something else.  

If you can't decide among the alternatives, I'd suggest that until "something better comes along" let's go with this till an Issue is filed due to an actual problem that we need fixed.

Reply:
maybe an explicit sentence to the effect of:

"When the chart is directly referencing properties of the spire config file, such as plugin names that are not easy to control, its acceptable to use the string verbatim rather then try and convert it into camel case in values."

There are likely better wordings then that.